core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -376,12 +378,25 @@ class Partition(val topic: String,
   }
 
   /**
+   * The low watermark offset value, calculated only if the local replica is the partition leader
+   * It is only used by leader broker to decide when DeleteRecordsRequest is satisfied. Its value is minimum logStartOffset of all live replicas
+   * Low watermark will increase when the leader broker receives either FetchRequest or DeleteRecordsRequest.
```

Comment:
Yes I think it is correct. For example, if there is only one replica, DeleteRecordsRequest or log retention will immediately increase LW.